dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPI.java

```diff
@@ -0,0 +1,84 @@
+package com.dotmarketing.factories;
+
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.VersionableAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.liferay.portal.model.User;
+
+public class MultiTreeAPI {
+
+    VersionableAPI vers = APILocator.getVersionableAPI();
+    ContentletAPI conAPI = APILocator.getContentletAPI();
+    User sys = APILocator.systemUser();
+    
+    
+    
+    public Table<String, String, Set<String>>  getPageMultiTrees(final IHTMLPage page, final boolean liveMode) throws DotDataException, DotSecurityException {
+        
+        Table<String, String, Set<String>> pageContents = HashBasedTable.create();
+        List<MultiTree> multis = MultiTreeFactory.getMultiTrees(page.getIdentifier());
+        for (MultiTree t : multis) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)